librz/analysis/arch_profile.c

```diff
@@ -19,15 +19,25 @@ RZ_API void rz_arch_profile_free(RzArchProfile *p) {
 	}
 	ht_up_free(p->registers_mmio);
 	ht_up_free(p->registers_extended);
+	free(p);
 }
 
 /**
  * \brief Creates a new RzArchProfile type
  */
 RZ_API RZ_OWN RzArchProfile *rz_arch_profile_new() {
 	RzArchProfile *profile = RZ_NEW0(RzArchProfile);
+	if (!profile) {
+		return NULL;
+	}
 	profile->registers_mmio = ht_up_new0();
+	if (!profile->registers_mmio) {
+		return NULL;
+	}
 	profile->registers_extended = ht_up_new0();
+	if (!profile->registers_extended) {
+		return NULL;
+	}
```

Comment:
you can't do this.
let's say that `profile->registers_mmio` is NULL.

by returning NULL you will not free `profile` which was allocated successfully previously.
you need to free also in those if.
for example 
```c
	if (!profile->registers_mmio) {
			free(profile);
			return NULL;
	}
	/// [....]
	if (!profile->registers_extended) {
			ht_up_free(profile->registers_mmio);
			free(profile);
			return NULL;
	}
```

Reply:
Oh. Okay, I'll fix that. Thanks for properly highlighting the erroneous code and for the example 💙